Add `[p]cleanup between`

NeuroAssassin opened the pull request:
### Type

- [ ] Bugfix
- [ ] Enhancement
- [X] New feature

### Description of the changes
This PR closes #2544; adds `[p]cleanup between <message id one> <message id two>` to cleanup messages between the two messages.

Commit:
Add `[p]cleanup between`

At the moment, only deletes the messages in between, does not delete the messages specified when running the command.

mikeshardmind commented on redbot/cogs/cleanup/cleanup.py:
```diff
@@ -271,6 +271,40 @@ def check(m):
+        try:
+            mone = await channel.fetch_message(one)
+        except discord.errors.Notfound:
+            return await ctx.send(f"Could not find a message with the ID of {one}.")
```
This needs to not use f-strings and use i18n

NeuroAssassin replied:
Addressed in 4bc6b5e0ff34cc458f92b7a05f3981e54ff9de48.

mikeshardmind commented on redbot/cogs/cleanup/cleanup.py:
```diff
@@ -271,6 +271,40 @@ def check(m):
+        try:
+            mtwo = await channel.fetch_message(two)
+        except discord.errors.Notfound:
+            return await ctx.send(f"Could not find a message with the ID of {two}.")
```
Same here on the f-strings and i18n

NeuroAssassin replied:
Addressed in 4bc6b5e0ff34cc458f92b7a05f3981e54ff9de48.

mikeshardmind commented on redbot/cogs/cleanup/cleanup.py:
```diff
@@ -271,6 +271,40 @@ def check(m):
+    @commands.guild_only()
+    @commands.bot_has_permissions(manage_messages=True)
+    async def between(
+        self, ctx: commands.Context, one: int, two: int, delete_pinned: bool = False
```
converters here should probably be changed to only match potentially valid snowflakes.

NeuroAssassin replied:
Addressed in 4bc6b5e0ff34cc458f92b7a05f3981e54ff9de48.

Commit:
Create converter for message IDs; remove f-strings; use converters in a few extra commands

Commit:
Sacrifice to the style gods

mikeshardmind commented on redbot/cogs/cleanup/cleanup.py:
```diff
@@ -271,6 +278,48 @@ def check(m):
+    ):
+        """Delete the messages between Messsage One and Message Two, providing the messages IDs.
+
+        The first message ID should be the older message, and the second one the newer.
```
We could remove the specific requirement of order here by just ordering them by which one was created first, but I actually like that this may help prevent mistakes.

mikeshardmind reviewed (dismissed):
:+1:

Commit:
small string fix

This should also get github to stop caring that it didnt get an old build result from Travis 3 days ago despite having a newer one
